client/client_test.go

```diff
@@ -41,12 +40,35 @@ func TestNew(t *testing.T) {
 	chainservB := chainservice.NewSimpleChainService(chain, b)
 	messageserviceB := messageservice.NewTestMessageService(b)
 	storeB := store.NewMockStore(bKey)
-	New(messageserviceB, chainservB, storeB, logDestination)
+	clientB := New(messageserviceB, chainservB, storeB, logDestination)
 
 	messageserviceA.Connect(messageserviceB)
 	messageserviceB.Connect(messageserviceA)
-	clientA.CreateDirectChannel(b, types.Address{}, types.Bytes{}, outcome.Exit{}, big.NewInt(0))
+	// Set up an outcome that requires both participants to deposit
+	outcome := outcome.Exit{outcome.SingleAssetExit{
+		Allocations: outcome.Allocations{
+			outcome.Allocation{
+				Destination: types.AddressToDestination(a),
+				Amount:      big.NewInt(5),
+			},
+			outcome.Allocation{
+				Destination: types.AddressToDestination(b),
+				Amount:      big.NewInt(5),
+			},
+		},
+	}}
 
-	<-time.After(time.Second)
+	id, res := clientA.CreateDirectChannel(b, types.Address{}, types.Bytes{}, outcome, big.NewInt(0))
+	got := <-res
+
+	if got.ObjectiveId != id {
+		t.Errorf("expected completed objective with id %v, but got %v", id, got.ObjectiveId)
+	}
+
+	gotFromB := <-clientB.ChannelFunded()
```

Comment:
It's good to see these two different ways that the client can signal to the application about the objective completing.

Maybe we ought to include just one method for now, seeing as the API probably needs a bit more planning / thought before we know what we need. How about we just keep the "big pipe that has everything" way for now, and drop the "promises" way until we decide we need it?

Reply:
[c20c46b](https://github.com/statechannels/go-nitro/pull/208/commits/c20c46b6cbf67ecfe2e264a8e6057fc39911ed9d)